tests/marker_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests_test
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubevirt/bridge-marker/tests"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("bridge-marker", func() {
+	Describe("bridge resource reporting", func() {
+		It("should be reported only when available on node", func() {
+
+			out, err := tests.RunOnNode("node01", "sudo ip link add br-test type bridge")
+			if err != nil {
+				panic(fmt.Errorf("%v: %s", err, out))
+			}
+
+			defer tests.RunOnNode("node01", "sudo ip link del br-test")
+
+			Eventually(func() bool {
+				node, err := clientset.CoreV1().Nodes().Get("node01", v1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				capacity, reported := node.Status.Capacity["bridge.network.kubevirt.io/br-test"]
+
+				if !reported {
+					return false
+				}
+				capacityInt, _ := capacity.AsInt64()
+				if capacityInt != int64(1000) {
+					return false
+				}
+				return true
+			}, 20*time.Second, 5*time.Second).Should(Equal(true))
+
+			out, err = tests.RunOnNode("node01", "sudo ip link del br-test")
+			if err != nil {
+				panic(fmt.Errorf("%v: %s", err, out))
+			}
+
+			Eventually(func() bool {
+				node, err := clientset.CoreV1().Nodes().Get("node01", v1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				_, reported := node.Status.Capacity["bridge.network.kubevirt.io/br-test"]
+				return reported
+
+			}, 20*time.Second, 5*time.Second).Should(Equal(false))
+		})
+	})
```

Comment:
another useful test would be creating a pod attached to the resource and checking that capacity is lowered / pod attached to the bridge.

Reply:
For bridges we don't actually have a set number of resources we can attach. We would have to come up with some magic number of maximum interfaces, check how many are connected, and display the result.
Even though we mark the capacity as 1k, it's more of a true/false value.

Reply:
You are right that we use is as true/false value. The main reason being that multi-net de-facto standard works with node resources.

However, kubernetes does used/free resources counting for us. It is just not shown in Node object (I believe the counter is hidden somewhere in scheduler to improve performance or some other good reason). i.e. If we attach 1000 pods to the bridge (way over the limit of pods per node), Kubernetes won't schedule more pods on the node.

Reply:
? Doesn't k8s expose counters on Node resource? There are `allocatable` counter and other counters too. I believe one of them is to track number of devices available to us.

Reply:
From https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/node-allocatable.md#allocatable

> Kubernetes nodes typically run many OS system daemons in addition to kubernetes daemons like kubelet, runtime, etc. and user pods. Kubernetes assumes that all the compute resources available, referred to as Capacity, in a node are available for user pods. In reality, system daemons use non-trivial amount of resources and their availability is critical for the stability of the system. To address this issue, this proposal introduces the concept of Allocatable which identifies the amount of compute resources available to user pods. Specifically, the kubelet will provide a few knobs to reserve resources for OS system daemons and kubernetes daemons.

There is no system pod using linux bridge, so Allocatable == Capacity

Reply:
@phoracek perhaps I am just bad at wording, but I meant to check that after a pod is created a resource is allocated (so `allocatable` would be decreased). I understand that `capacity` will stay the same.